### ISSUE TITLE
Sharing abicheck: deeper ABI/API scans with reusable baselines

### ISSUE THREAD

**Issue by napetrov** [lvc/abi-compliance-checker#137]
Thanks for maintaining ABI Compliance Checker — it remains a foundational tool in the C/C++ ABI compatibility ecosystem.

I wanted to share **[abicheck](https://github.com/abicheck/abicheck)**, a Python-native ABI/API scanner. For new CI setups, the recommended migration is to create a fresh abicheck baseline from a known-good release and use the native `scan` workflow rather than carrying an older descriptor or dump forward.

## Create a new baseline

Build the current known-good release with debug information, then capture its binary, public headers, build context, and sources:

```bash
pip install abicheck

abicheck dump build/libfoo.so \
  -H include/ \
  --build-info build/ \
  --sources . \
  --depth source \
  -o libfoo.abicheck.json
```

Store this JSON snapshot with the release or in your CI artifact storage. Advance it only when intentionally establishing a new compatibility baseline.

## Scan future builds

```bash
abicheck scan build/libfoo.so \
  -H include/ \
  --build-info build/ \
  --sources . \
  --depth source \
  --against libfoo.abicheck.json
```

`--depth source` enables the deepest scan: binary exports and metadata, debug-info layouts, public-header ASTs, actual build flags, source replay, and the reachability graph. This catches changes that symbol- or header-only checks cannot see, including build-mode drift, macro/default-argument changes, and inline/template body changes.

If an evidence source is unavailable, abicheck reports that coverage gap instead of silently treating the missing check as a pass.

## Scan results

The published benchmark counts unsupported inputs, errors, skips, and timeouts as misses:

| Scanner | Correct / 193 | Accuracy | False positives | False negatives | Time |
|---|---:|---:|---:|---:|---:|
| **abicheck + headers** | **185** | **95.9%** | **0** | 8 | 199 s |
| **abicheck + build/sources** | **192** | **99.5%** | **0** | 1 | 916 s |
| ABICC via `abi-dumper` | 86 | 44.6% | 8 | 99 | 872 s |
| ABICC XML mode | 78 | 40.4% | 7 | 108 | 1,871 s |

The source-aware scan is intentionally more expensive, but it provides the widest available coverage. Full methodology and per-case results: [Benchmark & Tool Comparison](https://abicheck.github.io/abicheck/reference/tool-comparison/).

abicheck supports ELF, PE/COFF, and Mach-O binaries, plus JSON, SARIF, Markdown, HTML, and JUnit reports. Header/source analysis requires a C/C++ toolchain; binary-only scans remain available when source evidence is not present.

Links: [repository](https://github.com/abicheck/abicheck) · [documentation](https://abicheck.github.io/abicheck/) · [scan guide](https://abicheck.github.io/abicheck/user-guide/scan-levels/) · [baseline guide](https://abicheck.github.io/abicheck/user-guide/baseline-management/)

Corrections and reproducibility feedback are welcome.
